appIcons.js

```diff
@@ -67,9 +67,11 @@ const MyAppIcon = new Lang.Class({
     Extends: AppDisplay.AppIcon,
 
     // settings are required inside.
-    _init: function(settings, app, iconParams, onActivateOverride) {
+    _init: function(settings, app, monitorNumber, iconParams, onActivateOverride) {
```

Comment:
little thing: maybe it should be named monitorIndex instead of monitorNumber, for consistency with the shell naming.

bigger thing. We are passing the monitor  number through all the classes. Could we retrieve the monitor internally with something like ```findMonitorForActor``` (https://git.gnome.org/browse/gnome-shell/tree/js/ui/layout.js#n891). I don't think it is a good idea (not sure if it it completely reliable for instance), but I wanted to point out the option.